src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2426,6 +2438,8 @@ def webapp_up(cmd, name, location=None, sku=None, dryrun=False, logs=False, laun
         # setting to build after deployment
         logger.warning("Updating app settings to enable build after deployment")
         update_app_settings(cmd, rg_name, name, ["SCM_DO_BUILD_DURING_DEPLOYMENT=true"])
+        # wait for all the settings to completed
+        time.sleep(30)
```

Comment:
Is it on purpose? 30 seconds is a long long time ....

Reply:
Yes for now since Kudu has a cold start of 20sec & when we do a app Recycle after setting an AppSetting. As a result of this we have seen that on first deployment the build never happens because we did not wait for the App Recycle to finish & the settings to be applied

Reply:
Also once we have this fixed on our backend - this will be considerably reduced